src/dataflow/src/render/reduce.rs

```diff
@@ -272,49 +274,41 @@ where
                                 // The inputs are triples of a reduction type, an optional index and row to decode.
                                 // The `reduce_abelian` operator guarantees that values will be placed in sorted order
                                 // by the values Ord implementation. This means that fused reductions
-                                // (ReductionType's FusedAccumulable and FusedMinMax) will be the first entries in inputs
+                                // (ReductionType's FusedAccumulable and FusedHierarchical) will be the first entries in inputs
                                 // if any such ReductionType's are present.
                                 // We need to reconstitute the final value by:
                                 // 1. Extracting out the fused aggregates
                                 // 2. For each aggregate we are computing, figure out if it is either a fused accumulable,
-                                //    fused min-max or unfused aggregate
+                                //    fused hierarchical or unfused aggregate
                                 // 3. Get the relevant value (either from the rows of fused aggregates, or from the input
                                 //    array for unfused aggregates)
                                 // 4. Stitch all the values together into one row.
 
                                 let new_row = Row::new(Vec::new());
+                                let mut accumulable = new_row.iter();
```

Comment:
is there a nicer idiom for "empty iterator"?

Reply:
You could use [`std::iter::empty()`](https://doc.rust-lang.org/std/iter/fn.empty.html)

Reply:
Ah, actually I don't think this will work because `empty()` returns an `Empty<T>` but you later assign a different type. There is `Datumlist::empty().iter()` which yields the same Iterator type

Reply:
thanks this was a great hint